### ISSUE TITLE
Notes on the world map

### ISSUE THREAD

**Issue by Kiaazad** [mapeditor/tiled#3136]
A tool to add notes to the world map can be pretty useful.
Right now the way to do it is adding a big or infinite empty map with an object layer. It can be hard to edit, a bit of hassle when trying to select other maps, and can be accidentally moved while moving something else.

**Comment by eishiya** [comment]
Maps and Worlds are two distinct entities in Tiled, "world map" makes it confusing which you mean. From context, sounds like you mean Worlds.

I think being able to add annotations to *any* Tiled document (Maps, Worlds, Tilesets, maybe even Projects somehow) could be useful though.

**Comment by bjorn** [comment]
Right, so first of all I can understand the problem. Worlds can only contain maps, so if you want to add any kind of annotation to the world, you need to do it as part of a map. Using an infinite map as part of your world just for annotations is an interesting idea, but indeed it comes with all kinds of usability issues since that was not how this feature was intended to be used.

I could imagine a number of solutions:

* Attempt to resolve the usability issues, making "infinite map on top of other maps in a world" a usable solution.
  * Would require improvements to the world tool, to work better with overlapping maps
  * Would require changes to make the moving of maps not so easy to trigger by accident
  * Probably requires ability to control Z order, since currently the "active map" always gets rendered on top, so you won't see the annotations for that map.

* Add explicit support for adding "note" objects to worlds.
  * I don't really like this, since it adds another data structure with required tooling / undo / scripting support.

* Attempt to unify the hierarchy, such that maps can be part of maps.
  * Would replace manually edited worlds with maps.
  * Would allow a single selection / movement tool to be used to position maps along with other objects.
  * Opens up many cans of worms, but I think it's the best long term solution.

**Comment by Kiaazad** [comment]
I support maps inside maps as well. We can do all kind of interesting stuff with the same tools we already have. And all it takes is the addition of a map layer type.
It doesn't have to be fancy either, I'll settle on clicking the layer, or a button next to it to edit it's map in another tab.
To be honest, when I first heard about worlds I expected a similar workflow to normal maps.
